R/list-simplify.R

```diff
@@ -0,0 +1,89 @@
+#' Simplify a list to an atomic or S3 vector
+#'
+#' @details
+#' Simplification maintains a one-to-one correspondence between the input
+#' and output, implying that each element of `x` must contain a vector of
+#' length 1. If you don't want to maintain this correspondence, then you
+#' probably want either [list_c()] or [list_flatten()].
+#'
+#' @param x A list.
+#' @param strict What should happen if simplification fails? If `TRUE`,
+#'   will error. If `FALSE`, will return `x` unchanged.
+#' @param ptype An optional prototype to ensure that the output type is always
+#'   the same.
+#' @returns A vector the same length as `x`.
+#' @export
+#' @examples
+#' list_simplify(list(1, 2, 3))
+#'
+#' try(list_simplify(list(1, 2, "x")))
+#' try(list_simplify(list(1, 2, 1:3)))
+list_simplify <- function(x, strict = TRUE, ptype = NULL) {
+  simplify_impl(x, strict = strict, ptype = ptype)
+}
+
+# Wrapper used by purrr functions that do automatic simplification
+list_simplify_internal <- function(
```

Comment:
Wrong indentation.

Reply:
I thought this was the double-indent style we reached agreement on in https://style.tidyverse.org/functions.html. I have mentally adjusted myself to like this style 😆/😭

Reply:
I think I may be getting used to it too. But I thought we'd only use it when the lines get too long?

Reply:
I think maybe we didn't mean the same thing by "lines that are too long", because it seems to me like it applies here because I can't otherwise fit the function definition on a single line.

Reply:
😱 😱 😱 😱 😱 😱 😱 😱 😱 😱 😱 😱